docs/models/blockstates/introduction.md

```diff
@@ -0,0 +1,52 @@
+Introduction to Blockstate JSONs
+================================
+
+Blockstate JSONs are Minecraft's way to map "variant strings" to models. A variant string can be absolutely anything, from "inventory" to "power=5" to "I am your father." They represent an actual model, where the blockstate is just a container for them. In code, a variant string within a blockstate JSON is represented by a `ModelResourceLocation`, normally shortened to just "MRL".
+
+When the game searches for a model corresponding to a block in the world, it takes the [blockstate][] for that position, and then it uses an `IStateMapper` to find the corresponding MRL for it, which then refers to the actual model. The default `IStateMapper` uses the block's registry name as the location of the blockstate JSON. (E.g. block `examplemod:testblock` goes to the RL `examplemod:testblock`.) The variant string is pieced together from the blockstate's properties. More information can be found [here][statemapper].
+
+As an example, let's take a look at the vanilla `oak_log.json`:
+
+```json
+{
+  "variants": {
+    "axis=y":  { "model": "oak_log" },
+    "axis=z":   { "model": "oak_log_side" },
+    "axis=x":   { "model": "oak_log_side", "y": 90 },
+    "axis=none":   { "model": "oak_bark" }
+  }
+}
+```
+
+Here we define 4 variant strings, and for each we use a certain model, either the upright log, the sideways log (rotated or not), and the all bark model (this model is not seen normally in vanilla; you have to use `/setblock` to create it). Since logs use the default `IStateMapper`, these variants will define the look of a log depending on the property `axis`.
+
+A blockstate always has to be defined for all possible variant strings. When you have many properties, this results in lots of possible variants, as every combination of properties must be defined. In Minecraft 1.8's blockstate format, you have to define every string explicitly, which leads to long, complicated files. It also doesn't support the concept of submodels, or multiple models in the same blockstate. In order to allievate this, Forge introduced its [own blockstate format][Forge blockstate], which is available in Minecraft 1.8 and up.
```

Comment:
I'm not sure how much of this is even yours, but this page should probably at least mention vanilla multipart blockstate files. Really this page should just mostly be a link to the MC wiki about models.